arangosh/Restore/RestoreFeature.cpp

```diff
@@ -816,13 +816,32 @@ arangodb::Result restoreView(arangodb::httpclient::SimpleHttpClient& httpClient,
 arangodb::Result triggerFoxxHeal(arangodb::httpclient::SimpleHttpClient& httpClient) {
   using arangodb::Logger;
   using arangodb::httpclient::SimpleHttpResult;
-  const std::string FoxxHealUrl = "/_api/foxx/_local/heal";
-
   std::string body = "";
 
+  // check if the foxx api is available.
+  const std::string statusUrl = "/_admin/status";
   std::unique_ptr<SimpleHttpResult> response(
-      httpClient.request(arangodb::rest::RequestType::POST, FoxxHealUrl,
+      httpClient.request(arangodb::rest::RequestType::POST, statusUrl,
                          body.c_str(), body.length()));
+  auto res =  ::checkHttpResponse(httpClient, response, "check status", body);
+  if(res.fail()) {
+    res.reset(res.errorNumber(), "While checking avilablility of foxx-api for heal: " + res.errorMessage());
+    return res;
+  }
+
+  auto builder = response->getBodyVelocyPack();
+  bool foxxApiEnabled = builder->slice().get("foxxApi").getBool();
```

Comment:
This should be handled a bit more gracefully. Consider the case when a devel/3.6 arangorestore is used against an ArangoDB 3.5, which does not export the `foxxApi` attribute. In this case the `getBool()` will throw an exception and the user will see a cryptic error message.
Let's please handle this gracefully. The attribute `foxxApi` should be considered optional, and if not present, we need to assume the Foxx API is turned on.